include/flamegpu/exception/FLAMEGPUException.h

```diff
@@ -420,6 +420,8 @@ DERIVED_FLAMEGPUException(InvalidDependencyGraph, "Agent function dependency gra
  * This should not occur if the shared ID matches ID_NOT_SET
  */
 DERIVED_FLAMEGPUException(AgentIDCollision, "Multiple agents of same type share an ID");
+DERIVED_FLAMEGPUException(IDCollision, "Generic ID collision");
+DERIVED_FLAMEGPUException(IDOutOfBounds, "Generic ID range failure");
```

Comment:
ID range will now be reported as CUDA error (though we could catch and rethrow).